crates/nu-cli/src/completions/completer.rs

```diff
@@ -262,6 +262,22 @@ impl NuCompleter {
                                 } else if prev_expr_str == b"ls" {
                                     let mut completer = FileCompletion::new();
 
+                                    return self.process_completion(
+                                        &mut completer,
+                                        &working_set,
+                                        prefix,
+                                        new_span,
+                                        fake_offset,
+                                        pos,
+                                    );
+                                } else if let Ok(true) = String::from_utf8(prev_expr_str).map(|x| {
+                                    x.ends_with('\"')
+                                        || x.ends_with('\'')
+                                        || x.parse::<f64>().is_ok()
+                                }) {
```

Comment:
Rolling your own flawed parser here makes no sense. We have more elaborate rules for string parsing etc. and it would redo work.

You already have access to the `FlatShape` of the `prev_expr` from above. There you can match on `String`/`Float` etc.

Reply:
Thanks for resolving this!